projects/packages/forms/src/blocks/contact-form/components/jetpack-integrations-modal/integration-card/integration-card-header.tsx

```diff
@@ -109,18 +108,24 @@ const IntegrationCardHeader = ( {
 							<span className="integration-card__description">{ description }</span>
 						) }
 						{ showPluginAction && (
-							<span className="integration-card__plugin-badge">
-								{ ! isInstalled && installPluginActionLabel }
-								{ isInstalled && ! isActive && activatePluginActionLabel }
-							</span>
+							<Badge
+								intent={ isInstalled && ! isActive ? 'warning' : 'default' }
+								className="integration-card__plugin-badge"
+							>
+								{ pluginActionLabel }
+							</Badge>
 						) }
 						{ showConnectedBadge && (
-							<span className="integration-card__connected-badge">
-								<Icon icon="yes-alt" size={ 12 } />
+							<Badge intent="success" className="integration-card__connected-badge">
 								{ __( 'Enabled', 'jetpack-forms' ) }
-							</span>
+							</Badge>
 						) }
-						{ showPendingBadge && <>{ pendingBadge }</> }
+						{ showPendingBadge &&
+							( setupBadge || (
+								<Badge intent="warning" className="integration-card__setup-badge">
+									{ __( 'Needs connection', 'jetpack-forms' ) }
+								</Badge>
+							) ) }
```

Comment:
I rebased this on top of some other work to [change the order of the badges on mobile](https://github.com/Automattic/jetpack/pull/44381). 

The main difference after addressing merge conflicts is we had to keep the class names like `integration-card__connected-badge` so that we could continue to apply the flex order changes on desktop vs mobile. 

Likewise, in the CSS we kept the rules applying the order for those classes.